Jenkinsfile

```diff
@@ -1,91 +1,117 @@
-                                        description: 'Repo for shared libraries',
-                                        name: 'LIBRARIES_REPO'
-                                ),
-                        ]
-                )
-        ]
+  [
+    parameters(
+      [
+        string(
+          defaultValue: 'x86_64,ppc64le',
+          description: 'A comma separated list of architectures to run the test on. Valid values include [x86_64, ppc64le, aarch64, s390x].',
+          name: 'ARCHES'
+        ),
+        string(
+          defaultValue: 'SSHPRIVKEY',
+          description: 'SSH private key Jenkins credential ID for Beaker/SSH operations',
+          name: 'SSHPRIVKEYCREDENTIALID'
+        ),
+        string(
+          defaultValue: 'SSHPUBKEY',
+          description: 'SSH public key Jenkins credential ID for Beaker/SSH operations',
+          name: 'SSHPUBKEYCREDENTIALID'
+        ),
+        string(
+          defaultValue: 'KEYTAB',
+          description: 'Kerberos keytab file Jenkins credential ID for Beaker/SSH operations',
+          name: 'KEYTABID'
+        ),
+        string(
+          defaultValue: 'v0.2',
+          description: 'Version of shared libraries to use',
+          name: 'LIBRARIES_VERSION'
+        ),
+        string(
+          defaultValue: 'https://github.com/RedHat-MultiArch-QE/multiarch-ci-libraries',
+          description: 'Repo for shared libraries',
+          name: 'LIBRARIES_REPO'
+        ),
+        string(
+          defaultValue: '',
+          description: 'Version of repo for tests to run.',
+          name: 'TEST_VERSION'
+        ),
+        string(
+          defaultValue: null
+          description: 'Repo for tests to run. If left blank, the current repo is assumed (*note* this default will only work for multibranch pipelines).',
+          name: 'TEST_REPO'
+        ),
+        string(
+          defaultValue: 'tests',
+          description: 'Directory containing tests to run. Should at least one of the follow: an ansible-playbooks directory containing one or more test directories each of which having a playbook.yml, a scripts directory containing one or more test directories each of which having a run-test.sh',
+          name: 'TEST_DIR'
+        )
+      ]
+    )
+  ]
 )
 
 List arches = params.ARCHES.tokenize(',')
 
 library changelog: false,
-        identifier: "multiarch-ci-libraries@${params.LIBRARIES_BRANCH}",
-        retriever: legacySCM([$class: 'GitSCM', branches: [[name: "*/${params.LIBRARIES_BRANCH}"]],
-                      doGenerateSubmoduleConfigurations: false,
-                      extensions: [], submoduleCfg: [],
-                      userRemoteConfigs:
-                              [[url: "${params.LIBRARIES_REPO}"]]])
+identifier: "multiarch-ci-libraries@${params.LIBRARIES_VERSION}",
+retriever: modernSCM([$class: 'GitSCMSource',remote: "${params.LIBRARIES_REPO}"])
```

Comment:
curious to see if this works when using a branch or tag.

Reply:
I will test this 👍

Reply:
Testing shows that this works for both branches and tags. It works similiar to how OpenShift git ref works when specify source for a template. I think I will change the name of the variables like this to reference to remain consistent with OpenShift.